### ISSUE TITLE
Failed to register SkyServe controller on non US Azure region

### ISSUE THREAD

**Issue by NicolasGrosjeanProbayes** [skypilot-org/skypilot#3260]
Hello,

When I did a `sky serve up`, after having only `francecentral` azure region into the catalog (like described in the [FAQ](https://skypilot.readthedocs.io/en/latest/reference/faq.html#advanced-how-to-make-skypilot-use-all-global-regions)), I succeed to provision on francecentral region, but when the job is submitted, I got an error saying the region is invalid.

Please find an extract of the logs

```
I 03-01 12:02:08 cloud_vm_ray_backend.py:1763] Launching on Azure francecentral
I 03-01 12:03:43 log_utils.py:45] Head node is up.
I 03-01 12:04:46 cloud_vm_ray_backend.py:1611] Successfully provisioned or found existing VM.
I 03-01 12:04:48 cloud_vm_ray_backend.py:4275] Processing file mounts.
I 03-01 12:04:48 cloud_vm_ray_backend.py:4307] To view detailed progress: tail -n100 -f ~/sky_logs/sky-2024-03-01-12-02-07-802222/file_mounts.log
I 03-01 12:04:48 backend_utils.py:1272] Syncing (to 1 node): /tmp/service-task-sky-service-569a-llfl2uf9 -> ~/.sky/serve/sky_service_569a/task.yaml.tmp
I 03-01 12:04:49 cloud_vm_ray_backend.py:3085] Running setup on 1 node.
I 03-01 12:04:58 cloud_vm_ray_backend.py:3098] Setup completed.
I 03-01 12:05:03 cloud_vm_ray_backend.py:3180] Job submitted with Job ID: 2

E 03-01 12:06:05 subprocess_utils.py:84] ValueError: Failed to register service 'sky-service-569a' on the SkyServe controller. Reason:
E 03-01 12:06:05 subprocess_utils.py:84] ValueError: Invalid region 'francecentral'
E 03-01 12:06:05 subprocess_utils.py:84] If a region is not included in the following list, please check the FAQ docs for how to fetch its catalog info.
E 03-01 12:06:05 subprocess_utils.py:84] https://skypilot.readthedocs.io/en/latest/reference/faq.html#advanced-how-to-make-skypilot-use-all-global-regions
E 03-01 12:06:05 subprocess_utils.py:84] List of supported azure regions: 'centralus, eastus, eastus2, northcentralus, southcentralus, westcentralus, westus, westus2, westus3'
E 03-01 12:06:05 subprocess_utils.py:84] 
E 03-01 12:06:05 subprocess_utils.py:84] 
RuntimeError: Failed to spin up the service. Please check the logs above for more details.
```

**Comment by concretevitamin** [comment]
Thanks for the report @NicolasGrosjeanProbayes! This seems like a bug we should fix: the local machine’s catalogs are not synced to the controller. Cc @cblmemo @Michaelvll

**Comment by romilbhardwaj** [comment]
Related - #3165

**Comment by romilbhardwaj** [comment]
Hey @NicolasGrosjeanProbayes - this will get fixed with #3262. 

If you'd like to try it before it gets merged, you can `git checkout controller_catalog_sync` and [install from source](https://skypilot.readthedocs.io/en/latest/getting-started/installation.html).

**Comment by NicolasGrosjeanProbayes** [comment]
Hey @romilbhardwaj ,

It works fine for me, thanks a lot.
